sympl/_core/constants.py

```diff
@@ -36,4 +37,7 @@
 
 
 def set_constant(name, value, units):
-    default_constants[name] = DataArray(value, attrs={'units': units})
+    if is_valid_unit(units):
+        default_constants[name] = DataArray(value, attrs={'units': units})
+    else:
+        raise TypeError('{} is not a valid unit.'.format(units))
```

Comment:
Should raise a ValueError, not a TypeError.

Reply:
Fixed